### DISCUSSION TITLE
Run in background

### DISCUSSION THREAD

**Discussion by RolfSievert** [ente-io/ente#668] [Enhancements]
Currently the photos app only does backup uploads while the app is open. I just joined and have 2k of photos to sync, which takes a while especially if I need to have the app opened. I have my default battery option in android set to "Optimized" which seem to interfere with ente running in the background.

It would be helpful if
- the app could run in the background with the optimized battery settings,
- or if not possible, identify this setting and notify that it has to be changed for the app to run in the background

I looked for this kind of setting in the app but couldn't find any, please tell me if I missed anything 😊 And thanks for providing an awesome alternative to google photos!

**Comment by vishnukvmd** [comment]
Hey, could you please long-press on the app's icon, click on `App info`-> `App battery usage` and choose `Unrestricted`?

This way you can whitelist Ente without altering your default system setting for other apps.

**Reply by RolfSievert** [reply]
That is a good solution in the meantime, thanks for the guidance (Android settings are hard to navigate...)!

**Reply by RolfSievert** [reply]
I still have issues with the app running in the background (even though it's set to `Unrestricted`, which seem to help a little at least). I believe that is because of limited RAM that Android stops apps running in the background. I have the same issues with all my other apps as well, so not a unique thing for Ente.

**Reply by mirh** [reply]
If you have a pixel or a sony, you just have to do that yes. But other OEMs are much much more to go through.
https://dontkillmyapp.com/

**Comment by jayjoshi64** [comment]
This is not an option for me in Pixel 7 (android 14). The `App battery usage` only has a single option called "Allow background usage" which is already on.   but the app still stops when switch to different apps.  
I have 3500 photos, and it has been 3 days and only 300 photos backed up so far.  Even worse is that for new photos, I have to open the app to start the syncing.
Let me know if there are any solutions for this.

**Reply by jayjoshi64** [reply]
I feel the app should register a background service or have an active notification to stay active like other apps do.

**Reply by kungfoome** [reply]
You have to actually click on "Allow background usage", so you turn it on and then click on the text that says "Allow background usage", then you can set unrestricted. Not sure why this is so hidden.

**Reply by 6058aa27-6ced-4d77-952a-bf55d59351c2** [reply]
> You have to actually click on "Allow background usage", so you turn it on and then click on the text that says "Allow background usage", then you can set unrestricted. Not sure why this is so hidden.

Jeez yeah that's hidden. Found it though. Now can it run in the background then?

Also agree that this should be more clear in the app. Some apps will prompt you for unrestricted battery, could try that. Or a persistent notification like another mentioned. Idk how this works really. But would be nice to sync in the background, especially when initially switching.
